Makefile

```diff
@@ -26,11 +26,17 @@ user-manual-pdf:
 	cd user_manual && make latexpdf
 	@echo "User manual build finished; PDF is updated"
 
-icons-docs: clean-icons-docs
+get-server-sources:
 	cd build && sh get-server-sources.sh $(DRONE_BRANCH)
+
+icons-docs: clean-icons-docs get-server-sources
 	cd build && composer install && composer update
 	cd build && php generateIconsDoc.php
 
+openapi-spec: get-server-sources
+	cd build/openapi-extractor && composer install
```

Comment:
I think we have a conflicting requirement for php versions now between
- build/composer.json (requiring 7.4)
- build/openapi-extractor/composer.json (requiring 8.0+)

We can probably just drop these two dependencies in build/composer.json as they are no longer used:

     11         "juliushaertl/phpdoc-to-rst": "dev-php7.0",
     12         "phpdocumentor/reflection": "dev-php7.0",



I noticed this when just running `make developer-manual-html` in the root, which is not catched by CI it seems. Maybe we can make sure that there is actually a github action triggering the icon-docs and openapi-spec makefile targets.

Reply:
Good catch. They are indeed not needed, but upgrading scssphp to a php 8.x version brings a lot of troubles :/ Still working on that...

Reply:
Even installing PHP7.4 and running the current code doesn't work for me...

Reply:
@juliushaertl I simply disabled generating the icons because it is broken anyway https://github.com/nextcloud/documentation/issues/7162. I don't like to leave it like that but for the tasks this seems to be the easiest because the icons-docs seem extremely broken.N ot only php7 and outdated dependencies, even with those fixed the output is complete garbage and there must be something else that is wrong.